Sources/SecureXPC/Client/XPCClient.swift

```diff
@@ -221,9 +223,35 @@ public class XPCClient {
         })
     }
 
-	// MARK: Abstract methods
+    private func getConnection() -> xpc_connection_t {
+        let newConnection = self.createConnection()
+
+        xpc_connection_set_event_handler(newConnection, self.handle(event:))
+        xpc_connection_resume(newConnection)
+
+        return newConnection
+    }
+
+    private func handle(event: xpc_object_t) {
+		// This callback is *not* called for "reply" messages,
+		// but *is* called when events happens when reply-less messages are sent
```

Comment:
So that means if an API user sets `errorHandler` then it's only called for reply-less messages? That seems like a deal breaker due to its inconsistent behavior. While the current behavior means issue #17 exists, it's very clear what the behavior is since there's no way to get back an error for the `send(...)` functions.

Seems like a way to ensure consistency here is under the hood to always send XPC messages that expect a reply and then have the `send(...)` functions that don't have a reply be able to provide an error handler. Or potentially as a bonus make it a completion handler so that it's possible for the client to get confirmation the message was received.

Aside from the consistency issues, the approach in this PR makes it impossible (or at least inconvenient?) for the client to know _which_ `send(...)` calls didn't succeed.

Reply:
Hmmm you're totally right.

> Seems like a way to ensure consistency here is under the hood to always send XPC messages that expect a reply and then have the send(...) functions that don't have a reply be able to provide an error handler. Or potentially as a bonus make it a completion handler so that it's possible for the client to get confirmation the message was received.

I experimented with this idea. One thing to note is that you can't just use a `Result<(), XPCError>` reply type as usual (because `Void` isn't `Codable`).

> Or potentially as a bonus make it a completion handler so that it's possible for the client to get confirmation the message was received.

I think this would be good. The completion handler can be `((XPCError?) -> Void)?`. When it's `nil`, we can optimize and use the reply-less API.

> Aside from the consistency issues, the approach in this PR makes it impossible (or at least inconvenient?) for the client to know which send(...) calls didn't succeed.

True. I didn't notice this because I was narrowly interested in seeing the connected/disconnected status of my connection (I built a small little debugging UI in my App using [`NSImageNameStatusNone`/`NSImageNameStatusAvailable`/`NSImageNameStatusUnavailable`](https://hetima.github.io/fucking_nsimage_syntax/) indicators)

Reply:
Actually, could we move this design-related discussion thread to the main issue? #17